frontend/packages/dev-console/src/components/catalog/utils/category-utils.ts

```diff
@@ -1,38 +1,37 @@
 import * as _ from 'lodash';
 import { CatalogItem } from '@console/plugin-sdk';
-import { CatalogCategories, CatalogCategory, CatalogSubcategory } from './types';
+import { CatalogCategory, CatalogSubcategory } from './types';
 
 export const matchSubcategories = (
   category: CatalogCategory,
   item: CatalogItem,
 ): (CatalogCategory | CatalogSubcategory)[] => {
   if (!category.subcategories) {
-    if (!category.values) {
+    if (!category.tags) {
       return [];
     }
 
-    let values = item[category.field];
-    if (!Array.isArray(values)) {
-      values = [values];
+    let { tags } = item;
+    if (!Array.isArray(tags)) {
+      tags = [tags];
     }
 
-    const intersection = [category.values, values].reduce((a, b) => a.filter((c) => b.includes(c)));
+    const intersection = [category.tags, tags].reduce((a, b) => a.filter((c) => b.includes(c)));
     if (!_.isEmpty(intersection)) {
       return [category];
     }
 
     return [];
   }
 
-  const matchedSubcategories = [];
-  _.forOwn(category.subcategories, (subCategory) => {
-    let values = item[category.field];
-
-    if (!Array.isArray(values)) {
-      values = [values];
+  const matchedSubcategories: (CatalogCategory | CatalogSubcategory)[] = [];
+  _.each(category.subcategories, (subCategory) => {
+    let { tags } = item;
+    if (!Array.isArray(tags)) {
```

Comment:
When is this not an array?

Reply:
Keeped the old code, cleaned it up and expect that it's now always an array (which we know for defaults and the CRD)